libraries/tracking/streams/product.js

```diff
@@ -47,7 +47,7 @@ const currentProductIdChanged$ = main$
  * Emits when the category route comes active again after a legacy page was active.
  */
 const productRouteReappeared$ = pwaDidAppear$
-  .filter(({ pathname }) => pathname.startsWith(ITEM_PATH));
+  .filter(({ action }) => action.route.pattern.startsWith(ITEM_PATH));
```

Comment:
Please always check pattern with an equal condition: `...pattern === '${ITEM_PATH}/:productId'`.
Exactly like you did in the search stream.

Reply:
Done